common/scala/build.gradle

```diff
@@ -18,19 +18,26 @@ dependencies {
     compile 'com.typesafe.akka:akka-http-core_2.11:10.0.10'
     compile 'com.typesafe.akka:akka-http-spray-json_2.11:10.0.10'
 
-    compile 'log4j:log4j:1.2.16'
+    compile 'ch.qos.logback:logback-classic:1.2.3'
+    compile 'org.slf4j:jcl-over-slf4j:1.7.25'
+    compile 'org.slf4j:log4j-over-slf4j:1.7.25'
     compile 'commons-codec:commons-codec:1.9'
     compile 'commons-io:commons-io:2.4'
-    compile 'commons-logging:commons-logging:1.2'
     compile 'commons-collections:commons-collections:3.2.2'
-    compile 'org.apache.zookeeper:zookeeper:3.4.10'
     compile 'org.apache.kafka:kafka-clients:0.10.2.1'
     compile 'org.apache.httpcomponents:httpclient:4.4.1'
     compile 'com.fasterxml.uuid:java-uuid-generator:3.1.3'
     compile 'com.github.ben-manes.caffeine:caffeine:2.4.0'
     compile 'com.google.code.findbugs:jsr305:3.0.2'
 }
 
+configurations {
+    compile.exclude group: 'log4j'
+    compile.exclude group: 'commons-logging'
+    runtime.exclude group: 'log4j'
+    runtime.exclude group: 'commons-logging'
+}
```

Comment:
Does it make sense to put this block on the offending libraries? e.g:

```groovy
compile('com.fasterxml.uuid:java-uuid-generator:3.1.3') {
    exclude group: 'log4j'
}
```

That'd make sure we definitly know what we exclude in the future and nothing breaks by accident?

Reply:
Definitely the better way. Done by amending the first commit